packages/calcite-components/src/components/dialog/dialog.tsx

```diff
@@ -309,12 +335,41 @@ export class Dialog
 
   @State() defaultMessages: DialogMessages;
 
+  @State() assistiveText: string | null = "";
```

Comment:
Should this be initialized to `null`? `updateAssistiveText` uses `null` when there's no text to display.